src/System.Net.Requests/tests/RequestStreamWrapper.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.IO;
+
+namespace System.Net.Tests
+{
+    internal class RequestStreamWrapper : IDisposable
+    {
+        private readonly HttpWebRequest _request;
+        private Stream _stream;
+
+        public Stream Stream
+        {
+            get
+            {
+                if (_stream == null)
+                    _stream = _request.GetRequestStream();
+                return _stream;
+            }
+        }
```

Comment:
I don't think this needs to be done lazily: the function to get one of these is calling GetRequestStream... it can be prompt and call _request.GetRequestStream in the ctor.